File: advent_of_code_2023/day9/part1.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

int res = 0;

int main(){
    int n;
    while(cin >> n){
        vector<int> arr[n+7];
        for(int i =0 ;i < n; i++){
            int a;
            cin >> a;
            arr[0].push_back(a);
        }
        for(int i = 1; i < n ; i++){
            for(int k = 0 ; k < arr[i-1].size()-1; k++){
                arr[i].push_back(arr[i-1][k+1] - arr[i-1][k]);
            }
        }
        for(int i = 0; i < n; i++){
            reverse(arr[i].begin(), arr[i].end());
        }
        arr[n-1].push_back(0);
        for(int i = n-2; i >= 0; i--){
            arr[i].push_back(arr[i][n-i-1] - arr[i+1][n-i-1]);
        }
        res += arr[0][n];
        for(int i = 0; i < n; i++){
            for(int k = 0; k < arr[i].size(); k++){
                cout << arr[i][k] << " ";
            }
            cout << endl;
        }
    }
    cout << res;
}

File: advent_of_code_2023/day14/part1.py
from sys import stdin

arr = []
res = 0

def Reverse(tup):
    new_tup = tup[::-1]
    return new_tup

for line in stdin:
    line = line[:-1]
    arr.append(line)

arr = list(zip(*arr[::-1]))

for i in range(len(arr)):
    arr[i] = Reverse(arr[i])

for line in arr:
    l = len(line)
    l2 = l
    for c in line:
        if c == 'O':
            res += l
            l-=1
        elif c == '#':
            l = l2-1
        l2 -= 1

print(res)